integration/container_stats_test.go

```diff
@@ -428,3 +428,78 @@ func testStats(t *testing.T,
 		require.NotEmpty(t, s.GetWritableLayer().GetInodesUsed().GetValue())
 	}
 }
+
+func TestContainerSysfsStatsWithPrivilegedPod(t *testing.T) {
+	if goruntime.GOOS == "windows" {
+		t.Skip("Doesn't care about filesystem properties on windows")
+	}
+	sb, sbConfig := PodSandboxConfigWithCleanup(
+		t,
+		"sandbox",
+		"sysfs-stats-with-privileged",
+		WithPodSecurityContext(true),
+	)
+
+	var (
+		testImage                 = images.Get(images.BusyBox)
+		containerName             = "test-container"
+		containerNameNoPrivileged = "test-container-no-privileged"
+	)
+
+	EnsureImageExists(t, testImage)
+	//sandbox and container with WithSecurityContext=true, sysfs is "ro"
+	cnConfig := ContainerConfig(
+		containerName,
+		testImage,
+		WithCommand("sh", "-c", "top"),
+		WithSecurityContext(true),
+	)
+	cn, err := runtimeService.CreateContainer(sb, cnConfig, sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.RemoveContainer(cn))
+	}()
+
+	t.Log("Start the container")
+	require.NoError(t, runtimeService.StartContainer(cn))
+	defer func() {
+		assert.NoError(t, runtimeService.StopContainer(cn, 10))
+	}()
+
+	t.Logf("Execute cmd in container by sync")
+	mountinfo, _, err := runtimeService.ExecSync(cn, []string{
+		"sh",
+		"-c",
+		"mount |grep sysfs",
+	}, 10)
+	assert.NoError(t, err)
+	assert.Contains(t, string(mountinfo), "rw")
+
+	//container with WithSecurityContext=false, sysfs is "ro"
+	cnConfigNoPrivileged := ContainerConfig(
+		containerNameNoPrivileged,
+		testImage,
+		WithCommand("sh", "-c", "top"),
+		WithSecurityContext(false),
+	)
+	cnNoPrivileged, err := runtimeService.CreateContainer(sb, cnConfigNoPrivileged, sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.RemoveContainer(cnNoPrivileged))
+	}()
+
+	t.Log("Start the container")
+	require.NoError(t, runtimeService.StartContainer(cnNoPrivileged))
+	defer func() {
+		assert.NoError(t, runtimeService.StopContainer(cnNoPrivileged, 10))
+	}()
+
+	t.Logf("Execute cmd in container by sync")
+	mountinfoNoPrivileged, _, err := runtimeService.ExecSync(cnNoPrivileged, []string{
+		"sh",
+		"-c",
+		"mount |grep sysfs",
+	}, 10)
+	assert.NoError(t, err)
+	assert.Contains(t, string(mountinfoNoPrivileged), "ro")
```

Comment:
nit: line 478 - 504 is largely a copy of line 451-476. I suggest we refactor a bit to reduce replication and make one test only test one scenario, something like (not well formatted):

```golang
func TestContainerSysfsStats(t *testing.T) {
    testcases := []struct{
      name string
      privilegedPod, privilegedContainer bool
      expectedSysfsOption string
    }
    {
      {"...", true, true, "rw"},
      {"...", false, false, "ro"},
      {"...", true, false, "ro"},
      {"...", false, true, "ro"},
    }

    for _, tt := range testcases {
      t.Run(tt.name, func(t *testing.T) {
        // your test logic, e.g., line 478 - 504
      }
    }
}
```